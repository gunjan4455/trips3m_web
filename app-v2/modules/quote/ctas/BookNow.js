import React, { Component } from 'react';
import PropTypes from 'prop-types';

import EVENTS_ENUMS from 'modules/quote/EventTypes';

class BookNow extends Component {

  handleBookNow = () => {
    const { tripId, quoteId } = this.props;
    this.props.callbackHandler(EVENTS_ENUMS.BOOK_NOW, { tripId, quoteId });
  };

  render() {
    const { isQuoteCardCta } = this.props;
    return (
      <div className="flexFull">
        <button className={isQuoteCardCta ? 'block p15 text-left wfull at_bookNow' : 'wfull btn-pri-large ripple pl5 pr5 at_bookNow'} onClick={this.handleBookNow}>
          Book Now
        </button>
      </div>
    );
  }
}

BookNow.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  opsDetail: PropTypes.object,
  callbackHandler: PropTypes.func,
  trackSegment: PropTypes.func,
  isQuoteCardCta: PropTypes.bool
};

BookNow.defaultProps = {
  opsDetail: {},
  callbackHandler: () => {
  },
  trackSegment: () => {
  },
  isQuoteCardCta: false
};

export default BookNow;
