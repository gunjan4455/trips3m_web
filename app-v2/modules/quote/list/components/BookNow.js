import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class BookNowPopup extends PureComponent {
  render() {
    const { agent, priceInfo, closeModal, bookNow } = this.props;

    return (
      <div className='pt24 pl15 pr15 pb24'>
        <div className='f14 sfc6 at_agentInfo'>
          {`You are booking quote by Agent ${agent.name} priced at ${priceInfo.price} (${priceInfo.perPerson === 'per person' ? 'per person' : ' total'}).`}
        </div>
        <div className='pt24 text-right'>
          <button className='mr24 sfc6 fw7 at_closeModal' onClick={closeModal}>Cancel</button>
          <button className="sfc1 fw7 at_continue" onClick={bookNow}>Continue</button>
        </div>
      </div>
    );
  }
}

BookNowPopup.propTypes = {
  agent: PropTypes.object,
  priceInfo: PropTypes.object,
  closeModal: PropTypes.func,
  bookNow: PropTypes.func
};

export default BookNowPopup;
