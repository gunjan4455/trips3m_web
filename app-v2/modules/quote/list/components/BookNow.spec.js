import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import BookNow from './BookNow';

configure({ adapter: new Adapter() });

const MockcloseModal = jest.fn();
const MockbookNow = jest.fn();

const DEFAULT_PROPS = {
  closeModal: MockcloseModal,
  bookNow: MockbookNow,
  agent: {
    name: 'xyz'
  },
  priceInfo: {
    price: 31500
  },
};

const shallowWrapperQuotesBox = (props) => {
  return shallow(<BookNow {...props} />);
};

describe('Quote module', () => {
  let props = DEFAULT_PROPS;

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperQuotesBox(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should closemodal function call ', () => {
    const wrapper = shallowWrapperQuotesBox(props);
    wrapper.find('.at_closeModal').simulate('click');
    expect(MockcloseModal).toHaveBeenCalled();
  });

  it('should bookNow function call ', () => {
    const wrapper = shallowWrapperQuotesBox(props);
    wrapper.find('.at_continue').simulate('click');
    expect(MockbookNow).toHaveBeenCalled();
  });

  it('should at_agentInfo class text correct', () => {
    const wrapper = shallowWrapperQuotesBox(props);
    expect(wrapper.find('.at_agentInfo').text())
    .toBe('You are booking quote by Agent xyz priced at 31500 ( total).')
  });

  it('should at_agentInfo class text correct when per person is defined in priceInfo', () => {
    props.priceInfo.perPerson = 'per person';
    const wrapper = shallowWrapperQuotesBox(props);
    expect(wrapper.find('.at_agentInfo').text())
    .toBe('You are booking quote by Agent xyz priced at 31500 (per person).')
  });

});
