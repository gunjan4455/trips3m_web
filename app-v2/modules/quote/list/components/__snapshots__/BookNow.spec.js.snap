// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookNowPopup
    agent={
      Object {
        "name": "xyz",
      }
    }
    bookNow={[MockFunction]}
    closeModal={[MockFunction]}
    priceInfo={
      Object {
        "price": 31500,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="f14 sfc6 at_agentInfo"
        >
          You are booking quote by Agent xyz priced at 31500 ( total).
        </div>,
        <div
          className="pt24 text-right"
        >
          <button
            className="mr24 sfc6 fw7 at_closeModal"
            onClick={[MockFunction]}
          >
            Cancel
          </button>
          <button
            className="sfc1 fw7 at_continue"
            onClick={[MockFunction]}
          >
            Continue
          </button>
        </div>,
      ],
      "className": "pt24 pl15 pr15 pb24",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You are booking quote by Agent xyz priced at 31500 ( total).",
          "className": "f14 sfc6 at_agentInfo",
        },
        "ref": null,
        "rendered": "You are booking quote by Agent xyz priced at 31500 ( total).",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="mr24 sfc6 fw7 at_closeModal"
              onClick={[MockFunction]}
            >
              Cancel
            </button>,
            <button
              className="sfc1 fw7 at_continue"
              onClick={[MockFunction]}
            >
              Continue
            </button>,
          ],
          "className": "pt24 text-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cancel",
              "className": "mr24 sfc6 fw7 at_closeModal",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Continue",
              "className": "sfc1 fw7 at_continue",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Continue",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="f14 sfc6 at_agentInfo"
          >
            You are booking quote by Agent xyz priced at 31500 ( total).
          </div>,
          <div
            className="pt24 text-right"
          >
            <button
              className="mr24 sfc6 fw7 at_closeModal"
              onClick={[MockFunction]}
            >
              Cancel
            </button>
            <button
              className="sfc1 fw7 at_continue"
              onClick={[MockFunction]}
            >
              Continue
            </button>
          </div>,
        ],
        "className": "pt24 pl15 pr15 pb24",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You are booking quote by Agent xyz priced at 31500 ( total).",
            "className": "f14 sfc6 at_agentInfo",
          },
          "ref": null,
          "rendered": "You are booking quote by Agent xyz priced at 31500 ( total).",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="mr24 sfc6 fw7 at_closeModal"
                onClick={[MockFunction]}
              >
                Cancel
              </button>,
              <button
                className="sfc1 fw7 at_continue"
                onClick={[MockFunction]}
              >
                Continue
              </button>,
            ],
            "className": "pt24 text-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cancel",
                "className": "mr24 sfc6 fw7 at_closeModal",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Continue",
                "className": "sfc1 fw7 at_continue",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Continue",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
